### ISSUE TITLE
PLUG_SETUPSTRUCT.hMenu changes after adding in plugin menu

### ISSUE THREAD

**Issue by mrfearless** [x64dbg/x64dbg#553]
Created 5 plugins: MenuTest1.dp32 - MenuTest5.dp32

Each plugin adds a single menu entry via plugsetup and calling:

```
plugsetup PROC PUBLIC USES EBX setupStruct:DWORD
    mov ebx, setupStruct
    ; Extract handles from setupStruct which is a pointer to a PLUG_SETUPSTRUCT structure  
    mov eax, [ebx].PLUG_SETUPSTRUCT.hwndDlg
    mov hwndDlg, eax
    mov eax, [ebx].PLUG_SETUPSTRUCT.hMenu
    mov hMenu, eax
    PrintText 'MenuTest #5'
    PrintDec hMenu
    mov eax, [ebx].PLUG_SETUPSTRUCT.hMenuDisasm
    mov hMenuDisasm, eax
    mov eax, [ebx].PLUG_SETUPSTRUCT.hMenuDump
    mov hMenuDump, eax
    mov eax, [ebx].PLUG_SETUPSTRUCT.hMenuStack
    mov hMenuStack, eax
    Invoke GuiAddLogMessage, Addr szPluginLoaded
    Invoke _plugin_menuaddentry, hMenu, MENU_PLUGIN, Addr szMenuPlugin
    ret
plugsetup endp
```

Checking the value of hMenu as each plugin is initialized it seems the hMenu value is changing on each plugin load. Adding MenuTest1, MenuTest2, and MenuTest3 show the appropriate menus fine - Adding MenuTest4 doesnt show any menu item in the plugins menu but shows in log that it loaded
![3menuplugins](http://s14.postimg.org/rjgzn9kdd/pluginmenu3showing.png)

And adding MenuTest5 to plugins folder causes x32dbg.exe to crash

With debugging output i could see that the hMenu was changing each load of a plugin:
![debugwindowofhmenu](http://s16.postimg.org/wl0vow091/pluginmenutest.png)

Might be my implementation of the plugin api calls, but wondered if someone could replicate this by creating 5 basic test plugins.dp32 that add a single menu item just to check if its an issue with my code or with x32dbg.exe (and possibly x64dbg.exe)

**Comment by mrexodia** [comment]
The hmenu are supposed to change for every plugin. The plugin menus are
sandboxed. Very weird though.

**Comment by mrfearless** [comment]
Here is the crash eventlog info:
```Faulting application name: x32dbg.exe, version: 0.0.2.5, time stamp: 0x56c39a02
Faulting module name: x32dbg.dll, version: 0.0.0.0, time stamp: 0x56c39a47
Exception code: 0xc0000409
Fault offset: 0x0006b420
Faulting process id: 0x748
Faulting application start time: 0x01d16cf4c0a0e445
Faulting application path: M:\x64dbg\x32\x32dbg.exe
Faulting module path: M:\x64dbg\x32\x32dbg.dll
Report Id: ff963f91-d8e7-11e5-81c1-08626648b163```

**Comment by mrfearless** [comment]
Decided to see what each structure item is as it was being loaded for the plugin, noticed the hwndDlg isnt consistent. Menu handles seem consistent up to MenuTest3 showing appropriate increment for hMenuDisasm, hMenuDump and hMenuStack, from MenuTest4, only the hMenuDump and hMenuStack seem correct, and MenuTest5 none are correct (except hwndDlg?) and causes crash.

Not sure why hwndDlg would change as its just getting this from GuiGetWindowHandle()

![MenuTestMoreInfo](http://s29.postimg.org/44id3jcon/menutestsmoreinfo.png)

**Comment by mrfearless** [comment]
Problem fixed. I created 5 x plugins via visual studio to test and it worked with them, so realised problem was with my sdk / code. After checking some code, realised that the pluginit, plugstop and plugsetup prototypes were not defined as 'C', was using STDCALL. Changed that and recompiled all test plugins and other plugins and seen issue was fixed. Updated my SDK to reflect newest snapshot release and those fixes for my code: https://bitbucket.org/mrfearless/x64dbg-plugin-sdk-for-masm
